File: DSA/Cpp/src/nodes.cpp
#include "nodes.h"

namespace dsa
{
    namespace nodes
    {
        Node::Node(int val) { this->val = val; }

        ListNode::ListNode(int val) : Node(val) {}
        ListNode::ListNode(int val, ListNode *next)
        {
            this->val = val;
            this->next = next;
        }

        TreeNode::TreeNode(int val) : Node(val) {}

        BinaryTreeNode::BinaryTreeNode(int val) : TreeNode(val) {}
        BinaryTreeNode::BinaryTreeNode(int val, BinaryTreeNode *left, BinaryTreeNode *right)
        {
            this->val = val;
            this->left = left;
            this->right = right;
        }

    }; // namespace nodes
} // namespace dsa

File: DSA/Python/src/nodes.py
class Node:
    def __init__(self, val: int = 0):
        self.val = val


class ListNode(Node):
    def __init__(self, val: int = 0, next=None):
        super().__init__(val)
        self.next = next


class TreeNode(Node):
    def __init__(self, val: int):
        super().__init__(val)


class BinaryTreeNode(TreeNode):
    def __init__(self, val: int, left=None, right=None):
        super().__init__(val)
        self.left = left
        self.right = right
